lib/srv/db/clickhouse/engine_native.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clickhouse
+
+import (
+	"context"
+	"crypto/tls"
+	"net/url"
+
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/lib/srv/db/common"
+	"github.com/gravitational/teleport/lib/utils"
+)
+
+func (e *Engine) handleNativeConnection(ctx context.Context, sessionCtx *common.Session) error {
+	u, err := url.Parse(sessionCtx.Database.GetURI())
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	tlsConfig, err := e.Auth.GetTLSConfig(ctx, sessionCtx)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	serverConn, err := tls.Dial("tcp", u.Host, tlsConfig)
```

Comment:
just a random thought, would a "creative" user be able to use clickhouse-native protocol to proxy something that's not a clickhouse database? i think as long the server can handle this `tlsConfig`, `tsh proxy db` basically becomes `tsh proxy app <tcp-app>`.

Reply:
> ? i think as long the server can handle this tlsConfig (e.g. serve TLS cert that matches the domain, ignore client cert), tsh proxy db basically becomes tsh proxy app <tcp-app>.

In tcp-app server just forward the traffic from the client to the app without any "internal" TLS Termination. In case of  ClickHouse native protocol the db engine will upgrade the user connection to TLS with using database certificates. So the flow is quite different.